myhoard/util.py

```diff
@@ -571,3 +571,18 @@ def run(self):
                 self.stats.unexpected_exception(ex=exception, where="RateTracker.run")
                 self.stats.increase("myhoard.ratetracker.errors")
                 self._reset()
+
+
+def restart_unexpected_dead_sql_thread(cursor, slave_status, stats, log):
+    assert slave_status["Slave_SQL_Running"] != "Yes"
```

Comment:
I don't think we should have this assert here.
Normally, if we read the code sequentially and assume no concurrent operations, it will always be true and harmless.
But if the thread is restarted externally between the ifs that guard the calls to  `restart_unexpected_dead_sql_thread` and this line, then we will get an `AssertionError`.
This error will not really enforce any invariant but it will be one more cause of failure, we lose more than we gain by failing here.

Reply:
I agree that normally this should always evaluate to true - but that should make it a great expression for an assert :)

Either way, I don't feel strongly about keeping it so removed.